### DISCUSSION TITLE
use SequenceEncoder without authenticating?

### DISCUSSION THREAD

**Discussion by Z-Rajaei** [pyg-team/pytorch_geometric#9040] [Q&A]
Hello,
I want to use the SequenceEncoder class to encode string attribute values as you have used in loading CSV for heterogeneous graphs.
But I get the error:
Repository Not Found for url: https://huggingface.co/api/models/sentence-transformers/name.
Please make sure you specified the correct `repo_id` and `repo_type`.
If you are trying to access a private or gated repo, make sure you are authenticated.
Invalid username or password.

Should I have a username and password? is there any way to use it free? Or what other encoding I can use to encode string values?

This is the class I tried to use:
```python
class SequenceEncoder:
    def __init__(self, model_name='all-MiniLM-L6-v2', device=None):
        self.device = device
        self.model = SentenceTransformer(model_name, device=device)

    @torch.no_grad()
    def __call__(self, df):
        x = self.model.encode(df.values, show_progress_bar=True,
                              convert_to_tensor=True, device=self.device)
        return x.cpu()
```

Any help is appreciated.

**Comment by rusty1s** [comment]
This is related to https://github.com/UKPLab/sentence-transformers/tree/master. Which version are you using? The README highlights the usage:
```python
from sentence_transformers import SentenceTransformer

model = SentenceTransformer("all-MiniLM-L6-v2")

sentences = [
    "This framework generates embeddings for each input sentence",
    "Sentences are passed as a list of string.",
    "The quick brown fox jumps over the lazy dog.",
]
sentence_embeddings = model.encode(sentences)

**Comment by Z-Rajaei** [comment]
Thank you,
It is solved.
I have some problems creating a graph. I refer to the movie example in your documentation to ask my question.
In the following movie example, the "title" and "genres" attribute values are encoded and then assigned to nodes. Is it possible to assign raw values to nodes? Why in NetworkX we assign attributes in this way: G.add_nodes_from([ (4, {"color": "red"}), (5, {"color": "green"})
I am confused about the encodings. Should the attributes be encoded or not? And if the name of attributes (like title or genres themselves) should be encoded or not?! Could you please clarify for me?

movie_x, movie_mapping = load_node_csv(
    movie_path, index_col='movieId', encoders={
        'title': SequenceEncoder(),
        'genres': GenresEncoder()
    })

**Reply by rusty1s** [reply]
In general, We want to hold "tensorfied"/encoded data in PyG that can be represented as tensors and act as neural network input. If you want to store raw values, please take a look at https://github.com/pyg-team/pytorch-frame.
